packages/astro/src/core/create-vite.ts

```diff
@@ -46,7 +46,7 @@ export async function createVite(inlineConfig: ViteConfigWithSSR, { astroConfig,
 		clearScreen: false, // we want to control the output, not Vite
 		logLevel: 'error', // log errors only
 		optimizeDeps: {
-			entries: ['src/**/*'], // Try and scan a user’s project (won’t catch everything),
+			entries: mode === 'dev' ? ['src/**/*'] : [], // Try and scan a user’s project (won’t catch everything),
```

Comment:
This greatly speeds up the build in large projects as Vite won't try to preoptimize. Want to make sure that this doesn't break anything though.

Reply:
I tested against demos and saw no difference with results, so I think this is ok to keep in.

Reply:
Worth doing `[]` in all cases?

Reply:
Would this break builds then (like if run in CI when dev hasn't been run)? If so happy to remove this, it helps get the build started quicker in large sites but if it might break things then it's not worth it.

Reply:
What about `src/pages/**/*.astro`? Would that give us most of the benefit while minimizing the potential for a bottleneck?

Reply:
Ok, I'll remove this part.